astropy/cosmology/connect.py

```diff
@@ -237,8 +246,8 @@ class CosmologyToFormat(io_registry.UnifiedReadWrite):
         Keyword arguments passed through to data writer.
     """
 
-    def __init__(self, instance, cls):
+    def __init__(self, instance: Cosmology, cls: type[Cosmology]):
         super().__init__(instance, cls, "write", registry=convert_registry)
 
-    def __call__(self, format, *args, **kwargs):
+    def __call__(self, format: str, *args: Any, **kwargs: Any) -> object:
```

Comment:
I don't think I've ever encountered `object` as a return type annotation, and at first glance, it feels much stricter than needed. Can you elaborate ?

Reply:
Agreed. I think this should be changed to `All`. `object` can be useful for input annotations. Less so for output ones.